tests/testthat/test-return_linter.R

```diff
@@ -882,3 +882,117 @@ test_that("return_linter lints `message`, `warning` and `stopifnot`", {
     linter
   )
 })
+
+test_that("non-if returns are skipped under allow_implicit_else = FALSE", {
+  expect_lint(
+    trim_some("
+      foo <- function(bar) {
+        bar
+      }
+    "),
+    NULL,
+    return_linter(allow_implicit_else = FALSE)
+  )
+})
+
+test_that("if/else don't throw a lint under allow_implicit_else = FALSE", {
+  expect_lint(
+    trim_some("
+      foo <- function(bar) {
+        if (TRUE) {
+          return(bar)
+        } else {
+          return(NULL)
+        }
+      }
+    "),
+    NULL,
+    return_linter(allow_implicit_else = FALSE)
```

Comment:
Isn't this a false negative?
I'd expect `return_style = "implicit"` to require
```r
if (TRUE) {
  bar
} else {
  NULL
}
```

Reply:
I think with `"implicit"` it should be allowed to use `return`s in control statements, especially in `if` and `switch`.  Maybe instead of more flags for control, one could introduce even more types like `“strict_implicit”` in which this would also be checked.

This would also fit with the original idea of ​​this PR to bring in the functionality as a new type.

Reply:
> I think with "implicit" it should be allowed to use returns in control statements

Disagree with this, and it's not the style followed by Tidyverse or base R. I agree it looks like a false negative to me, investigating. We can file a request to support mixed implicit/explicit returns as a separate FR.

Reply:
Corrected this test, and added the incorrect test as a new regression test.

That said, the new fix is more expedient than perfect -- it'll throw an incorrect lint here:

```r
foo <- function() {
  if (TRUE) {
    if (TRUE) {
      return(1)
    }
    2
  } else {
    3
  }
```

(the `return()` is a valid early return within that branch).

But as this issue is not relevant to the implicit `else` returns feature, I'm pushing to follow-up.